### PR TITLE
Refactor driver

### PR DESCRIPTION
**Pull request by versatran01**
WIP. Probably needs some review before merging.

### CODE REVIEW

File: src/imu_vn_100.cpp [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+void ImuVn100::FixImuRate() {
+  if (imu_rate_ <= 0) {
+    ROS_WARN("Imu rate %d is < 0. Set to %d", imu_rate_, kDefaultImuRate);
+    imu_rate_ = kDefaultImuRate;
+  }
+
+  if (kBaseImuRate % imu_rate_ != 0) {
+    int imu_rate_old = imu_rate_;
+    // TODO: THIS DOENS'T DO WHAT'S INTENDED
+    imu_rate_ = kBaseImuRate / (kBaseImuRate / imu_rate_old);
```
Comment (andre-nguyen) [inline src/imu_vn_100.cpp]: What's the meaning of that comment?

File: launch/vn_100_cont.launch [hunk 2]
```diff
@@ -1,40 +1,31 @@
...
-
-    <!-- Frame ID for messages -->
     <arg name="frame_id" default="$(arg imu)"/>
-
-    <!-- Baudrate of serial comms (see manual for allowed values) -->
     <arg name="baudrate" default="921600"/>
-
-    <!-- Data rate in Hz -->
-    <arg name="imu_rate" default="200"/>
+    <arg name="imu_rate" default="100"/>
```
Comment (andre-nguyen) [inline launch/vn_100_cont.launch]: Maybe add the binary output enabled param here, just for completeness. Either this or in the readme. Admittedly thoughI can't see why someone would want to get data in ascii.

File: README.md [hunk 3]
```diff
@@ -36,43 +36,39 @@ The frame ID entry for the sent messages.
...
-enable_temperature    (bool, default: true)
+enable_magn  (bool, default: true)
+enable_press (bool, default: true)
+enable_temp  (bool, default: true)
 ```
 
 Enable other possible messages that the driver is available to send. Note that the frequency of the data for each of these messages will be the same with the IMU data, if the topic is enabled.
 
-`enable_sync_out` (`bool`, `true`)
+`sync_out_rate` (`int`, `20`)
```
Comment (andre-nguyen) [inline README.md]: Is it `sync_out_rate` or `sync_rate`? And related to this, are you using an external clock to trigger cameras and time stamp the IMU or are you using the sync out of the imu to trigger the cameras?

Comment (versatran01) [inline README.md]: I changed it to `sync_rate`. We are using the sync out of the imu to trigger the cameras.

**Pull request merged**